cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go

```diff
@@ -1323,13 +1323,10 @@ func TestNodeGroupTemplateNodeInfo(t *testing.T) {
 				cpuKey:      "2",
 				gpuTypeKey:  gpuapis.ResourceNvidiaGPU,
 				gpuCountKey: "1",
+				labelsKey:   "kubernetes.io/arch=arm64,my-custom-label=custom-value",
```

Comment:
> this looks like a good ugprade, thanks for the merge request @aleskandro . i have one suggestion though, could we add some unit tests to ensure that this works the way we want and so that we don't regress in the future.

Hi @elmiko, thanks for the review. Do you think it is good to patch one of the current cases to also consider the labels set via the capacity annotations or were you meant to have a dedicated new test case?

(The `nodeLabels` map I deleted seems not needed when `includeNodes` is false, is it?)

Reply:
i'm ok with either as long as we have all the conditions of the new logic tested for both positive and negative assertions.